.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,26 +1,62 @@
 
-### Tests
+## Contribution Checklist
 
-- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
+  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/AVRO/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
+  
+  - Name the pull request in the form "AVRO-XXXX: [component] Title of the pull request", where *AVRO-XXXX* should be replaced by the actual issue number. 
+    The *component* is optional, and but can help identify the correct reviewers faster: either the language ("java", "python") or subsystem such as "build" or "doc" are good candidates.  
 
-### Commits
+  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
+  
+  - Make sure that the change passes the automated tests. You can [build the entire project](https://github.com/apache/avro/blob/master/BUILD.md) or just the [language-specific SDK](https://avro.apache.org/project/how-to-contribute/#unit-tests).
 
-- [ ] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
-  1. Subject is separated from body by a blank line
-  1. Subject is limited to 50 characters (not including Jira issue reference)
-  1. Subject does not end with a period
-  1. Subject uses the imperative mood ("add", not "adding")
-  1. Body wraps at 72 characters
-  1. Body explains "what" and "why", not "how"
+  - Each pull request should address only one issue, not mix up code from multiple issues.
+  
+  - Each commit in the pull request has a meaningful commit message (including the JIRA id)
 
-### Documentation
+  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.
 
-- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
-  - All the public functions and the classes in the PR contain Javadoc that explain what it does
+  - Every commit message references Jira issues in their subject lines. In addition, commits follow the guidelines from [How to write a good git commit message](https://chris.beams.io/posts/git-commit/)
+    1. Subject is separated from body by a blank line
+    1. Subject is limited to 50 characters (not including Jira issue reference)
+    1. Subject does not end with a period
+    1. Subject uses the imperative mood ("add", not "adding")
+    1. Body wraps at 72 characters
+    1. Body explains "what" and "why", not "how"
+
+-->
+
+## What is the purpose of the change
+
+*(For example: This pull request improves file read performance by buffering data.)*
```

Comment:
May add here explicit JIRA link if exists.

Reply:
I wish we could click on the JIRA in the title of the PR!  Since that's not the case, it's nice to have the direct link in the text.

Reply:
It's configured here and it definitely works in Pull Request description text, which is nice!

But it's not automatic in PR titles, and it's hit and miss in commit message text (one out of two are autolinked on this page: https://github.com/apache/avro/pull/1851/commits/7736b690bc111a254396557233ef067611255fd1)